util/movement-types/src/block.rs

```diff
@@ -103,8 +112,12 @@ impl Block {
 		)
 	}
 
-	pub fn add_transaction(&mut self, transaction: Transaction) {
+	pub fn add_transaction(&mut self, transaction: Transaction) -> Result<(), BlockError> {
+		if self.transactions.len() >= MAX_TRANSACTIONS {
```

Comment:
There's no max Tx per block but more a max size of the Vec<TX> array. Tx size can be very different, and we don't want that a block is rejected by the Celestia Da because of the size.